### PR TITLE
SAS-5 Dependency security checks

### PR DESCRIPTION
**Pull request by hx-markterry**
https://holidayextras.jira.com/browse/SAS-5

#### What does this PR do? (please provide any background)

Adds snyk checks into makeup

#### What tests does this PR have?

Added new unit tests to check that streams are correctly returned from the integrations

#### How can this be tested?

Run the following to see the new check in action:

    bin/make-up.js

#### What gif best describes how you feel about this work?

giphy

---

- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** \*
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,5 +22,10 @@ makeUp.check = function(options, callback) {
 };
 
 makeUp._runIntegration = function(options, item, callback) {
-  item.run(options, process.stdout, callback);
+  var intStream = new streams.WritableStream();
+  intStream.write('\n' + item.label + '\n==============\n');
+  item.run(options, intStream, function(err, result) {
+    console.log(intStream.toString());
```
Comment (hx-markterry) [inline index.js]: This will still mess up the output, like 2 ghostbusters crossing their streams.

**Comment by hx-markterry** [comment]
From ollie "for your use case, I'd just build a new Buffer(), wait for data on the stream, concat it into the buffer, then on the `end` event, flush it to the console

should only be 5 lines of code :simple_smile:"

**Comment by jameshhx** [comment]
Taking :one:

**Comment by jameshhx** [comment]
:+1: here

**Comment by djbeaumont** [comment]
Taking :two:

File: index.js [hunk 2]
```diff
@@ -15,9 +18,18 @@ makeUp.path = function(item) {
...
+    }, '');
+    callback(error, result);
+  });
 };
 
 makeUp._runIntegration = function(options, item, callback) {
-  item.run(options, process.stdout, callback);
+  var intStream = new streams.WritableStream();
+  intStream.write('\n' + item.label + '\n==============\n');
+  item.run(options, intStream, function(err, result) {
```
Comment (djbeaumont) [inline index.js]: Would this work as just `item.run(options, intStream, callback)` ?

File: index.js [hunk 3]
```diff
@@ -15,9 +18,18 @@ makeUp.path = function(item) {
...
+    var result = allStreams.reduce(function(previous, stream) {
+      return previous + stream.toString();
+    }, '');
+    callback(error, result);
+  });
 };
 
 makeUp._runIntegration = function(options, item, callback) {
-  item.run(options, process.stdout, callback);
+  var intStream = new streams.WritableStream();
```
Comment (djbeaumont) [inline index.js]: Does int stand for integration here? My first thought was 'why are they writing strings into a stream of integers?' :)

File: lib/integrations/snyk.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+var snykReport = require('snyk-report');
+
+var SnykIntegration = module.exports = {};
+
+SnykIntegration.label = 'SNYK Report';
+
+SnykIntegration.run = function(options, output, callback) {
+  snykReport('.', function(err, result) {
+    if (err) return callback(err, output);
```
Comment (djbeaumont) [inline lib/integrations/snyk.js]: Could a tiny bit of duplication be removed by changing this to `if (!err) output.write(...);` ?
Also, is it worth having a test for this file?

**Comment by djbeaumont** [comment]
Looks great :shipit:

**Pull request merged**